internal/rpcserver/interface.go

```diff
@@ -437,3 +438,28 @@ type FeeEstimator interface {
 	// high degree of certainty.
 	EstimateFee(targetConfs int32) (dcrutil.Amount, error)
 }
+
+// BlockTemplater represents a block templater for use with the RPC server.
+//
+// The interface contract requires that all of these methods are safe for
+// concurrent access.
+type BlockTemplater interface {
+	// ForceRegen asks the block templater to generate a new template immediately.
+	ForceRegen()
+
+	// Subscribe subscribes a client for block template updates.  The returned
+	// template subscription contains functions to retrieve a channel that produces
+	// the stream of block templates and to stop the stream when the caller no
+	// longer wishes to receive new templates.
+	Subscribe() *mining.TemplateSubscription
+
+	// CurrentTemplate returns the current template associated with the block
+	// templater along with any associated error.
+	CurrentTemplate() (*mining.BlockTemplate, error)
+
+	// UpdateBlockTime updates the timestamp in the passed header to the current
+	// time while taking into account the median time of the last several
```

Comment:
Since this is an interface, I would suggest not calling out the specifics of what it takes into account, rather "while taking into account the chain consensus rules".